pkg/netpol/connlist/parsed_resources_test.go

```diff
@@ -0,0 +1,1304 @@
+	// }
+}
+
+func stringForEval(peer eval.Peer) string {
+	if peer.IsPeerIPType() {
+		ipBlock, _ := eval.PeerIPToIPBlock(peer)
+		return ipBlock.ToCidrListString()
+	} else {
+		return peer.Namespace() + "/" + peer.Name()
+	}
+}
+
+func runParsedResourcesEvalTests(t *testing.T, testList []ParsedResourcesTest) {
+	t.Helper()
+	for i := 0; i < len(testList); i++ {
+		test := &testList[i]
+		t.Run(test.Name, func(t *testing.T) {
+			t.Parallel()
+			test.InitParsedResourcesTest()
+			if test.Banp != nil { // Tanya - remove this 'if' whenever BaselineAdminNetworkPolicy is implemented
+				return // Skip tests with BANP, until implemented
+			}
+			pe, err := eval.NewPolicyEngineWithObjects(test.Resources)
+			require.Nil(t, err, test.TestInfo)
+			peerList, err := pe.GetPeersList()
+			require.Nil(t, err, test.TestInfo)
+			peers := make([]Peer, len(peerList))
+			for i, p := range peerList {
+				peers[i] = p
+			}
+			connsRes := make([]Peer2PeerConnection, 0)
+			for _, src := range peerList {
+				for _, dst := range peerList {
+					if (src.String() == dst.String()) || (src.IsPeerIPType() && dst.IsPeerIPType()) {
+						continue
+					}
+					allowedConns, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(src, dst)
+					require.Nil(t, err, test.TestInfo)
+
+					srcForEval := stringForEval(src)
+					dstForEval := stringForEval(dst)
+					contProtocol, contPort := allowedConns.PickContainedConn()
+					if contPort != "" {
+						var res bool
+						// Tanya: uncomment whenever CheckIfAllowed supports ANPs
+						// res, err := pe.CheckIfAllowed(srcForEval, dstForEval, contProtocol, contPort)
+						// require.Nil(t, err, test.TestInfo)
+						// require.Equal(t, true, res, test.TestInfo)
+						res, err = pe.CheckIfAllowedNew(srcForEval, dstForEval, contProtocol, contPort)
+						require.Nil(t, err, test.TestInfo)
+						require.Equal(t, true, res, test.TestInfo)
+					}
+					uncontProtocol, uncontPort := allowedConns.PickUncontainedConn()
+					if uncontPort != "" {
+						var res bool
+						// Tanya: uncomment whenever CheckIfAllowed supports ANPs
+						// res, err := pe.CheckIfAllowed(srcForEval, dstForEval, uncontProtocol, uncontPort)
+						// require.Nil(t, err, test.TestInfo)
+						// require.Equal(t, false, res, test.TestInfo)
+						res, err = pe.CheckIfAllowedNew(srcForEval, dstForEval, uncontProtocol, uncontPort)
+						require.Nil(t, err, test.TestInfo)
+						require.Equal(t, false, res, test.TestInfo)
+					}
+
+					if allowedConns.IsEmpty() {
+						continue
+					}
+
+					p2pConnection := &connection{
+						src:               src,
+						dst:               dst,
+						allConnections:    allowedConns.AllConnections(),
+						protocolsAndPorts: allowedConns.ProtocolsAndPortsMap(),
+					}
+					connsRes = append(connsRes, p2pConnection)
+				}
+			}
+			connsFormatter, err := getFormatter(test.OutputFormat, peers)
+			require.Nil(t, err, test.TestInfo)
+			out, err := connsFormatter.writeOutput(connsRes)
```

Comment:
`eval` pkg will not support calling `connlist` funcs such the formatter's `writeOutput`;
so I suggest to do same expected connection check for `AllAllowedConnectionsBetweenWorkloadPeers` as for `CheckIfAllowedNew`.

Reply:
Same as above